lib/puppet/indirector/rest.rb

```diff
@@ -52,6 +52,10 @@ def self.port
     Puppet::Util::Connection.determine_port(port_setting, server_setting)
   end
 
+  def use_http_client?
+    Puppet.runtime['http'].instance_of?(Puppet::HTTP::Client)
```

Comment:
This needs to take into account `Puppet::Network::HttpPool.http_client_class` until puppetserver registers its own httpclient.

Reply:
Done